test/unit/test_compare.js

```diff
@@ -0,0 +1,233 @@
+                   test: "test1"
+               })
+        });
+    });
+
+    describe("$scope.addProject", function () {
+
+        beforeEach(function() {
+            $scope = {};
+            $attrs = {};
+            $scope.$evalAsync = function() {}
+            controller = $controller('CompareController', {
+                $scope: $scope,
+                $attrs: $attrs,
+            });
+        });
+
+        it('checks the value of seletedProjects if the project already exists',
+           function() {
+               var project = {id: 1}
+               var selected = [{id: 1}, {id: 2}]
+               $scope.selectedProjects = selected
+               $scope.addProject(project)
+               expect($scope.selectedProjects).toEqual(selected)
+        });
+
+        it('checks the value of seletedProjects and showProgress',
+           function() {
+               var project = {id: 1}
+               $scope.selectedProjects = [{id: 2}]
+               $scope.selectedTest = {}
+               $scope.selectedSuite = {}
+               $scope.addProject(project)
+               expect($scope.selectedProjects).toEqual([{id: 2}, {id: 1}])
+               expect($scope.showProgress[project.id]).toBe(true)
+               expect($scope.showResults).toBe(true)
+        });
+    });
+
+    describe("$scope.removeProject", function () {
+
+        beforeEach(function() {
+            $scope = {};
+            $attrs = {};
+            $scope.$evalAsync = function() {}
+            controller = $controller('CompareController', {
+                $scope: $scope,
+                $attrs: $attrs,
+            });
+        });
+
+        it('checks the value of seletedProjects after removal',
+           function() {
+               var project = {id: 1}
+               $scope.selectedProjects = [{id: 1}, {id: 2}]
+               $scope.removeProject(project)
+               expect($scope.selectedProjects).toEqual([{id: 2}])
+        });
+    });
+
+
+    describe("$scope.loadMoreData", function () {
+
+        var $httpBackend;
+
+        beforeEach(function() {
+            $scope = {};
+            $attrs = {};
+            $location = "";
+            controller = $controller('CompareController', {
+                $scope: $scope,
+                $attrs: $attrs,
+                $location: $location
+            });
+        });
+
+        beforeEach(inject(function($injector) {
+            $httpBackend = $injector.get('$httpBackend');
+            // Ignore requests from select2 for now.
+            $httpBackend.whenGET(/.*?api\/projects?.*/g).respond(
+                200, []);
+
+            $httpBackend.whenGET(/.*?project_test_results?.*/g).respond(
+                200, ["data"]);
+
+            // Ignore rest of select 2 requests.
+            $httpBackend.whenGET(/.*?api\/suitemetadata?.*/g).respond(
+                200, {results: [{id: "1"}]});
+            $httpBackend.whenGET(/.*?api\/suitemetadata?.*/g).respond(
```

Comment:
Duplicated line?

Reply:
Yes, that's because there are 2 requests to suitemetadata in compare.js which we need to both wait for and return dummy data (or nothing).